R/module_init_data.R

```diff
@@ -72,25 +72,27 @@ srv_init_data <- function(id, data, modules, filter = teal_slices()) {
     data_validated <- if (inherits(data, "teal_data_module")) {
       srv_teal_data(
         "teal_data_module",
-        data = reactive(req(FALSE)), # to .fallback_on_failure to shiny.silent.error
+        data = reactive(teal_data()), # to .fallback_on_failure to shiny.silent.error
         data_module = data,
         modules = modules,
         validate_shiny_silent_error = FALSE
       )
     } else if (inherits(data, "teal_data")) {
       reactiveVal(data)
     } else if (test_reactive(data)) {
-      .fallback_on_failure(this = data, that = reactive(req(FALSE)), label = "Reactive data")
+      data
     }
 
     if (inherits(data, "teal_data_module")) {
       shinyjs::disable(selector = sprintf(".teal-body:has('#%s') .nav li a", session$ns("content")))
     }
 
     observeEvent(data_validated(), {
-      showNotification("Data loaded successfully.", duration = 5)
-      shinyjs::enable(selector = sprintf(".teal-body:has('#%s') .nav li a", session$ns("content")))
-      if (isTRUE(attr(data, "once"))) {
+      # showNotification("Data loaded successfully.", duration = 5)
+
+      req(inherits(data_validated(), "teal_data"))
+
```

Comment:
```suggestion
      req(inherits(data_validated(), "teal_data"))
      showNotification("Data loaded successfully.", duration = 5)
```

Reply:
I will need to move this notification to another place. And, also rename some variables as they are misleading now.

I think it's best to place data type assertions in every reactive to hint the code reader which data object types is handled. In this case `data_validated` can also pass errors which are converted into `reactive(teal_data())` later so, we will need to show to notification there.

Reply:
After some testing I will finalize the state of the data flow and it will be easy to do this by referring [this diagram](https://app.diagrams.net/#G1PzwKjZHaST7Nnv62MWa2wvcBjphmlnSF#%7B%22pageId%22%3A%22EZsrWMlh8mfaOJpvtUJ1%22%7D).